### DISCUSSION TITLE
why rabbitmq_prometheus not support the producers and consumers of queue metrics

### DISCUSSION THREAD

**Discussion by jiangxinlingdu** [rabbitmq/rabbitmq-server#4998] [Other]
As of 3.8.0, RabbitMQ ships with built-in Prometheus & Grafana support.

why rabbitmq_prometheus not support the producers and consumers of queue metrics

**Comment by mkuratczyk** [comment]
Hi,

What specific metrics are you looking for? Have you seen
https://www.rabbitmq.com/prometheus.html#metric-aggregation ?
Perhaps you just need to enable/use per-object metrics.

Best,

On Wed, Jun 8, 2022 at 10:58 AM 匠心零度 ***@***.***> wrote:

> As of 3.8.0, RabbitMQ ships with built-in Prometheus & Grafana support.
>
> why rabbitmq_prometheus not support the producers and consumers of queue
> metrics
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rabbitmq/rabbitmq-server/issues/4996>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACI7PIXF7PPNXWB5Y542B33VOBOBVANCNFSM5YFXWR5Q>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Michał
RabbitMQ team

**Comment by jiangxinlingdu** [comment]
3q

**Comment by jiangxinlingdu** [comment]
<img width="1285" alt="image" src="https://user-images.githubusercontent.com/6373972/172582583-f0747860-0667-444c-8826-30fb38036111.png">
if i has  queues : q1、q2、q3  

how to find  q1  queue producers and consumers  metrics 
and how to find q2、and q3 queues  producers and consumers  metrics ？？？

**Comment by jiangxinlingdu** [comment]
<img width="1228" alt="image" src="https://user-images.githubusercontent.com/6373972/172584914-94ebf07b-d98b-490d-a752-574a3e86779e.png">

**Comment by mkuratczyk** [comment]
I guess there is just no metric for this right now, I doubt there is any other reason for that than "it was not implemented so far". You can get channel stats to get some of this data currently:
```
curl -s http://localhost:15692/metrics/per-object | grep ^rabbitmq_queue_messages_published_total
rabbitmq_queue_messages_published_total{channel="<0.974.0>",queue_vhost="/",queue="q1",exchange_vhost="/",exchange="direct"} 5
rabbitmq_queue_messages_published_total{channel="<0.1000.0>",queue_vhost="/",queue="q3",exchange_vhost="/",exchange="direct"} 5
rabbitmq_queue_messages_published_total{channel="<0.987.0>",queue_vhost="/",queue="q2",exchange_vhost="/",exchange="direct"} 5
```

**Comment by jiangxinlingdu** [comment]
3q  
how to find queues  consumers metrics ？？？

**Comment by michaelklishin** [comment]
The plugin does support per-object metrics if you [enable them or use a dedicated endpoint](https://www.rabbitmq.com/prometheus.html#metric-aggregation). This can lead to extremely
large responses when there's a lot of objects in the system.

What specifically are the metrics you are looking for?

**Reply by jiangxinlingdu** [reply]
see producers and consumers of queue metrics，dimension of the queue

**Reply by jiangxinlingdu** [reply]
i want see such as:

rabbitmq_queue_messages_published_total{queue_vhost="/",queue="q1",exchange_vhost="/",exchange="direct"} 5
rabbitmq_queue_messages_published_total{queue_vhost="/",queue="q3",exchange_vhost="/",exchange="direct"} 5
rabbitmq_queue_messages_published_total{queue_vhost="/",queue="q2",exchange_vhost="/",exchange="direct"} 5





rabbitmq_queue_messages_delivered_total{queue_vhost="/",queue="q1",exchange_vhost="/",exchange="direct"} 5
rabbitmq_queue_messages_delivered_total{queue_vhost="/",queue="q3",exchange_vhost="/",exchange="direct"} 5
rabbitmq_queue_messages_delivered_total{queue_vhost="/",queue="q2",exchange_vhost="/",exchange="direct"} 5



rabbitmq_queue_messages_confirmed_total{queue_vhost="/",queue="q1",exchange_vhost="/",exchange="direct"} 5
rabbitmq_queue_messages_confirmed_total{queue_vhost="/",queue="q3",exchange_vhost="/",exchange="direct"} 5
rabbitmq_queue_messages_confirmed_total{queue_vhost="/",queue="q2",exchange_vhost="/",exchange="direct"} 5


but per-object metrics not find consumers of queue metrics
